docs/parcels.md

```diff
@@ -0,0 +1,109 @@
+# Parcels
+
+A single-spa parcel is a bundle of functionality (like an application or component) meant to be controlled imperatively 
+by an application, Parcels use similar methodology as Applications, and can be considered an advanced single spa feature so we recommend [starting with Applications.](/docs/applications.md)
+A parcel can be as large as an application or as small as an component and written in 
+any language as long as it exports the correct lifecycle events. In a single-spa world, your SPA contains 
+many registered applications and potentially many parcels. Usually you'll want to mount a parcel within the 
+context of an application, this will automatically unmount the parcel when the application is unmounted.
+
+## Parcel Lifecycles
+To read more about lifecycles and how single-spa handles the concept start with [applications](/docs/applications.md#registered-application-lifecycle)
+
+### Bootstrap
+This lifecycle function will be called once, right before the parcel is
+mounted for the first time.
+
+```js
+function bootstrap(props) {
+  return Promise
+    .resolve()
+    .then(() => {
+        // This is where you do one-time initialization
+        console.log('bootstrapped!')
+        });
+}
+```
+
+### Mount
+This lifecycle function will be called whenever the parcel is not mounted and `mountParcel` is called. When
+called, this function should create DOM elements, DOM event listeners, etc. to render content to the user.
+
+```js
+function mount(props) {
+  return Promise
+    .resolve()
+    .then(() => {
+        // This is where you tell a framework (e.g., React) to render some ui to the dom
+        console.log('mounted!')
+        });
+}
+```
+
+### Unmount
+This lifecycle function will be called whenever the parcel is mounted, and either A) `unmount()` is called 
+or B) the parent parcel or Application is unmounted. When called, this function should clean up all DOM elements, 
+DOM event listeners, leaked memory, globals, observable subscriptions, etc. that were created at any point when the parcel was mounted.
+
+```js
+function unmount(props) {
+  return Promise
+    .resolve()
+    .then(() => {
+        // This is where you tell a framework (e.g., React) to unrender some ui from the dom
+        console.log('unmounted!');
+        });
+}
+```
+
+### Update (optional)
+This lifecycle function will be called whenever the parcel is mounted, and it needs to be updated.
```

Comment:
This seems like an important function, will there be a sample like above?

Reply:
This sentence is misleading. It says that the update lifecycle "will be called whenever the parcel is mounted," which is not true. Instead, it will only be called when the user of the parcel calls `parcel.update()`. I know that you are trying to explain that it gets called while the parcel is in a MOUNTED status, but I don't think that came across.

> The update lifecycle function will be called whenever the user of the parcel calls `parcel.update()`. Since this lifecycle is optional, the user of a parcel needs to check whether the parcel has implemented the update lifecycle before attempting to call.